src/main/java/fi/aalto/cs/apluscourses/ui/base/TreeView.java

```diff
@@ -151,9 +160,12 @@ public void valueChanged(@NotNull TreeSelectionEvent e) {
 
     @Override
     public void mouseClicked(@NotNull MouseEvent e) {
```

Comment:
do you think the logic inside the method can be tested fully or maybe at least to the extent of the conditional statements?